x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeService.java

```diff
@@ -193,16 +193,21 @@ private class ExchangeTransportAction implements TransportRequestHandler<Exchang
         @Override
         public void messageReceived(ExchangeRequest request, TransportChannel channel, Task task) {
             final String exchangeId = request.exchangeId();
-            ActionListener<ExchangeResponse> listener = new ChannelActionListener<>(channel);
+            ActionListener<ExchangeResponse> closingListener = new ChannelActionListener<>(channel).delegateFailure(
+                (listener, response) -> {
+                    listener.onResponse(response);
+                    response.decRef();
```

Comment:
This seems like the wrong place for a `decRef()` to me - we're letting go of the page in `org.elasticsearch.compute.operator.exchange.ExchangeSinkHandler#notifyListeners`, shouldn't we `defRef()` there instead? At the moment this is the only listener to consider AFAICT, but it seems that might not be true forever, and we'll be requiring _every_ listener to do the requisite decref to avoid leaks.

Reply:
++ lets dec ref where we actually instantiate the object @alex-spies

Reply:
Agree, makes sense!